pkg/databind/internal/secrets/command.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package secrets
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+)
+
+type Command struct {
+	Path           string   `yaml:"path"`
+	Args           []string `yaml:"args,omitempty"`
+	PassthroughEnv []string `yaml:"passthrough_environment,omitempty"`
+}
+
+type commandGatherer struct {
+	cfg *Command
+}
+
+// Error handling.
+var (
+	ErrNoPath                 = errors.New("secrets gatherer command must have a path parameter in order to be executed")
+	ErrEmptyResponse          = errors.New("the command returned an empty response")
+	ErrInvalidResponse        = errors.New("the command returned an invalid response")
+	ErrParseResNoData         = errors.New("missing required field 'data'")
+	ErrParseResInvalidData    = errors.New("invalid type for field 'data'")
+	ErrParseResTTLInvalidType = errors.New("invalid type for field 'ttl'")
+)
+
+func validationError(err error) error {
+	return fmt.Errorf("validation error: %w", err)
+}
+
+func runCommandError(err error) error {
+	return fmt.Errorf("failed to run command: %w", err)
+}
+
+func parseCmdResponseError(err error) error {
+	return fmt.Errorf("failed to parse command response: %w", err)
+}
+
+func invalidTypeError(t interface{}) error {
+	return fmt.Errorf("invalid type: %T", t)
+}
+
+func commandExitError(exitErr *exec.ExitError) error {
+	return fmt.Errorf("%s: %s", exitErr.Error(), string(exitErr.Stderr))
+}
+
+// End error handling.
+
+type cmdResponse struct {
+	CmdTTL  string         `json:"ttl,omitempty"`
+	CmdData map[string]any `json:"data"`
+}
+
+// Custom unmarshaler for cmdResponse.
+// The top-level field "data" is required, but the field "ttl" is optional.
+func (c *cmdResponse) UnmarshalJSON(data []byte) error {
+	// Top-level field "data" is required
+	genericRes := map[string]any{}
+	if err := json.Unmarshal(data, &genericRes); err != nil {
+		return err
+	}
+
+	if _, ok := genericRes["data"]; !ok {
+		return parseCmdResponseError(ErrParseResNoData)
+	}
+	// The nested data field must be either a string or a map[string]any.
+
+	d, err := stringOrMapStringAny(genericRes["data"])
+	if err != nil {
+		return parseCmdResponseError(ErrParseResInvalidData)
+	}
+
+	c.CmdData = d
+	if ttl, ok := genericRes["ttl"]; ok {
+		if s, ok := ttl.(string); ok {
+			c.CmdTTL = s
+		} else {
+			return parseCmdResponseError(ErrParseResTTLInvalidType)
+		}
+	}
+
+	return nil
+}
+
+func (c cmdResponse) TTL() (time.Duration, error) {
```

Comment:
Not a big issue in this scope, but I would follow the advise of being consistent with the method receiver ☝️ 

>Next is consistency. If some of the methods of the type must have pointer receivers, the rest should too, so the method set is consistent regardless of how the type is used. See the section on [method sets](https://go.dev/doc/faq#different_method_sets) for details.

https://go.dev/doc/faq#methods_on_values_or_pointers